main.cpp

```diff
@@ -15,7 +15,12 @@ New/This/Pointers/References conclusion
  */
 
 
+struct A {};
 
+struct HeapA
+{
+    A* aPtr = new A;
+};
```

Comment:
What happens when an instance of `HeapA` goes out of scope?
Does that `A` that you allocated on the heap leak?  Does it get destroyed correctly? 
- [ ] revise

Reply:
> What happens when an instance of HeapA goes out of scope?

All member variables will be destroyed.

> Does that A that you allocated on the heap leak? Does it get destroyed correctly?

Yes because after destroying the `aPtr` pointer you cant access to the instance of `A`. I have to write a deconstructor to release the memory via `delete`. The pointer must be resetted with `nullptr`.